OCI docker push fails

## Reproduce

As described in the [Deploy serverless application to OCI guide](https://guides.micronaut.io/latest/micronaut-oracle-function-maven-java.html):
* Create the application with [`mn create-function-app --build=maven --jdk=17 --lang=java --test=junit --features=graalvm,oracle-function com.example.demo`](https://micronaut.io/launch/?type=FUNCTION&name=demo&package=com.example&javaVersion=JDK_17&lang=JAVA&build=MAVEN&test=JUNIT&features=graalvm&features=oracle-function&version=3.8.7)
* Set the jib docker image in the pom file: `<jib.docker.image>[REGION].ocir.io/[TENANCY]/[REPO]/${project.artifactId}</jib.docker.image>`
* Run `docker login ....ocir.io` and then run `./mvnw deploy -Dpackaging=docker`

## Error

```
[ERROR] Failed to execute goal com.google.cloud.tools:jib-maven-plugin:3.1.4:dockerBuild (default-dockerBuild) on project demo: Tried to pull image manifest for registry-1.docker.io/fnproject/fn-java-fdk:latest but failed because: OCI index found, but accept header does not support OCI indexes: 404 Not Found
[ERROR] {"errors":[{"code":"MANIFEST_UNKNOWN","message":"OCI index found, but accept header does not support OCI indexes"}]}
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```
Full log: [log.txt](https://github.com/micronaut-projects/micronaut-maven-plugin/files/11023279/log.txt), stacktrace: [stacktrace.txt](https://github.com/micronaut-projects/micronaut-maven-plugin/files/11023419/stacktrace.txt).

Seems like the issue is when pulling `fn-java-fdk` and some kind of oci accept header is missing in the request. But when I run `docker pull fnproject/fn-java-fdk:latest`, it works fine. 

Not sure if the problem is similar to this one: https://github.com/fraunhoferfokus/deckschrubber/issues/43

--- a/parent/build.gradle
+++ b/parent/build.gradle
@@ -49,7 +49,7 @@ ext.extraPomInfo = {
         'azure-functions-maven-plugin.version'('1.5.0')
         'exec-maven-plugin.version'('1.6.0')
         'function-maven-plugin.version'('0.9.8')
-        'jib-maven-plugin.version'('3.1.4')
+        'jib-maven-plugin.version'('3.3.1')
         'maven-compiler-plugin.version'('3.10.1') // Override actual Maven compiler version (3.1) because some bugs cause annotation processors doesn't work well
         'maven-deploy-plugin.version'('3.0.0')
         'maven-failsafe-plugin.version'('2.22.2') // Override actual Maven surefire and failsafe version (2.12) to get native support for executing tests on the JUnit Platform (JUnit 5)

